src/Microsoft.IIS.WebManager/app/files/file-editor.ts

```diff
@@ -94,9 +94,10 @@ export class FileEditor {
         this._dirty = false;
 
         this.getFileText().then(txt => {
-            this._text.next(txt);
-            this._dirty = false;
-           
+            if (txt) {
```

Comment:
es6 more rigorous type checking, txt might be null

Reply:
Isn't this a behavioral change though? Previously the subject was able to emit null now it isn't. What if the user wants to delete all text then save?